From 2411a4536a2b75edde164a00ea650d6488bed7e3 Mon Sep 17 00:00:00 2001
From: "Ruiqiang.Hao" <Ruiqiang.Hao@windriver.com>
Date: Thu, 31 Oct 2019 13:45:07 +0800
Subject: [PATCH 386/386] Fix Serror panic by add mdelay.

There will be a Serror when the kernel starts.
Add mdelay(5) in function advk_pcie_check_pio_status.

Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/pci/controller/pci-aardvark.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pci/controller/pci-aardvark.c b/drivers/pci/controller/pci-aardvark.c
index 9143a0fe4cd2..a566a6101119 100644
--- a/drivers/pci/controller/pci-aardvark.c
+++ b/drivers/pci/controller/pci-aardvark.c
@@ -362,6 +362,7 @@ static void advk_pcie_check_pio_status(struct advk_pcie *pcie)
 	char *strcomp_status, *str_posted;
 
 	reg = advk_readl(pcie, PIO_STAT);
+	mdelay(5);
 	status = (reg & PIO_COMPLETION_STATUS_MASK) >>
 		PIO_COMPLETION_STATUS_SHIFT;
 
-- 
2.17.1

